src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -107,6 +110,94 @@ protected int read(ByteBuffer bb)
         return ChannelInputStream.read(ch, bb, true);
     }
 
+    @Override
+    public byte[] readAllBytes() throws IOException {
+        if (!(ch instanceof SeekableByteChannel))
+            return super.readAllBytes();
+        SeekableByteChannel sbc = (SeekableByteChannel)ch;
+
+        long length = sbc.size();
+        long position = sbc.position();
+        long size = length - position;
+
+        if (length <= 0 || size <= 0)
+            return super.readAllBytes();
+
+        if (size > (long) Integer.MAX_VALUE) {
+            String msg =
+                String.format("Required array size too large: %d = %d - %d",
+                    size, length, position);
+            throw new OutOfMemoryError(msg);
+        }
+
+        int capacity = (int)size;
+        byte[] buf = new byte[capacity];
+
+        int nread = 0;
+        int n;
+        for (;;) {
```

Comment:
```suggestion
        while (true) {
```

Reply:
The use of `for(;;)` is common; not changed.